### PR TITLE
Parent joint idx issue 6080

### PR DESCRIPTION
**Pull request by wayne-chen**
- Fix to default parentjointindex property of entities to -1 (INVALID_JOINT_INDEX) if unspecified.

https://highfidelity.manuscript.com/f/cases/6080/Default-parentjointindex-value-of-65535-doesn-t-make-sense

### CODE REVIEW

**Comment by wayne-chen** [comment]
To test:

- Launch interface
- In menu, select Settings->Advanced Menus
- In menu, select Settings->Developer Menus
- In menu, open Edit->Console
- Run code in the Console similar to what is shown below to create an entity:
```
var entity = Entities.addEntity({type: "Box", position: MyAvatar.position, rotation:MyAvatar.orientation, dimensions: {x:1,y:1,z:1}});
```
- Then, get the parentjointindex of the generated entity:
```
console.log(JSON.stringify(Entities.getEntityProperties(entity)));
```
- The parentjointindex should be -1 since the code posted does not specify the parentjointindex to set for the generated entity.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9914/HighFidelity-Beta-PR13195-6e13cad99aa0a509525a9dcd173dfe418001eb09.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9914/HighFidelity-Beta-Interface-PR13195-6e13cad99aa0a509525a9dcd173dfe418001eb09.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9914/HighFidelity-Beta-PR13195-6e13cad99aa0a509525a9dcd173dfe418001eb09.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9914/HighFidelity-Beta-Interface-PR13195-6e13cad99aa0a509525a9dcd173dfe418001eb09.exe

File: libraries/octree/src/OctreePacketData.cpp [hunk 1]
```diff
@@ -326,6 +326,18 @@ bool OctreePacketData::appendColor(colorPart red, colorPart green, colorPart blu
...
+    const unsigned char* data = (const unsigned char*)&value;
+    
+    int length = sizeof(value);
+    bool success = append(data, length);
+    if (success) {
+        _bytesOfValues += length;
+        _totalBytesOfValues += length;
+    }
+    return success;
+}
```
Comment (AndrewMeadows) [inline libraries/octree/src/OctreePacketData.cpp]: Technically this PR makes a change to the message protocol, however perhaps it doesn't matter... as long as no one is using legit joint indices larger than 32768?  Except for -1, of course.

I guess the way to test would be:
**(1)** Startup two domains, one using this PR and the other using `hifi/master` (or some recently merged PR).
**(2)** UserA uses Interface from this PR and UserB uses interface from `hifi/master` (or some recently merged PR).
**(3)** If both UserA and UserB can "hold" (aka "near-grab") objects in both domains and the other user sees everything work correctly... then they are compatible and no protocol version bump would be required.

File: libraries/shared/src/SpatiallyNestable.h [hunk 2]
```diff
@@ -26,7 +26,7 @@ using SpatiallyNestableWeakConstPointer = std::weak_ptr<const SpatiallyNestable>
 using SpatiallyNestablePointer = std::shared_ptr<SpatiallyNestable>;
 using SpatiallyNestableConstPointer = std::shared_ptr<const SpatiallyNestable>;
 
-static const uint16_t INVALID_JOINT_INDEX = -1;
+static const int16_t INVALID_JOINT_INDEX = -1;
```
Comment (AndrewMeadows) [inline libraries/shared/src/SpatiallyNestable.h]: I prefer `int16_t` over `qint16`.  Not a blocker but: perhaps we could do this for all changes to this PR?

**Comment by wayne-chen** [comment]
This PR propagates into another issue with `EntityItem::computeCollisionAndFinalMask`.  There is an access violation with writing to parameter `mask`.  I'm not sure where the root cause resides in.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9927/HighFidelity-Beta-PR13195-45c251b06146822df43ac84233fa719a4c4dca8e.apk

**Comment by wayne-chen** [comment]
Steps to repro this problem:

1.  Launch Interface with debugging
2.  Enter HMD mode
3.  Write access exception should be thrown from `EntityItem::computeCollisionGroupAndFinalMask`.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9927/HighFidelity-Beta-PR13195-45c251b06146822df43ac84233fa719a4c4dca8e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9927/HighFidelity-Beta-Interface-PR13195-45c251b06146822df43ac84233fa719a4c4dca8e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9927/HighFidelity-Beta-PR13195-45c251b06146822df43ac84233fa719a4c4dca8e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9927/HighFidelity-Beta-Interface-PR13195-45c251b06146822df43ac84233fa719a4c4dca8e.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10691/HighFidelity-Beta-PR13195-9a3c986.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10691/HighFidelity-Beta-PR13195-9a3c986.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10691/HighFidelity-Beta-Interface-PR13195-9a3c986.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11172/HighFidelity-Beta-PR13195-9a336f6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11172/HighFidelity-Beta-Interface-PR13195-9a336f6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11172/HighFidelity-Beta-PR13195-9a336f6.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11172/HighFidelity-Beta-Interface-PR13195-9a336f6.exe

**Comment by ZappoMan** [comment]
closed - super old

**Pull request closed without merging**